--- origsrc/libatomic_ops-1.2/configure.ac	2005-09-28 16:29:49.000000000 -0500
+++ src/libatomic_ops-1.2/configure.ac	2007-07-31 09:37:13.320056600 -0500
@@ -3,7 +3,7 @@
 AC_CANONICAL_TARGET([])
 AC_CONFIG_SRCDIR(src/atomic_ops.c)
 AM_INIT_AUTOMAKE
-AC_PROG_RANLIB
+AM_PROG_LIBTOOL
 
 AM_CONFIG_HEADER(src/config.h)
 
@@ -18,43 +18,25 @@
 AC_HEADER_STDC
 # AC_CHECK_HEADERS([ ])
 
-# Determine PIC flag, adjust default CFLAGS
+# Adjust default CFLAGS
 need_asm=false
-PICFLAG=
-AC_MSG_CHECKING(Determining PIC compiler flag)
-if test "$GCC" = yes; then
-  AC_MSG_RESULT(-fPIC)
-  PICFLAG=-fPIC
-else
   case "$host" in
     *-*-hpux*)
-      AC_MSG_RESULT("+Z")
-      PICFLAG="+Z"
       if test "$GCC" != yes; then
         CFLAGS="$CFLAGS +O2 -mt"
       fi
       ;;
     *-*-solaris*)
-      AC_MSG_RESULT(-Kpic)
-      PICFLAG=-Kpic
       if test "$GCC" != yes; then
         CFLAGS="$CFLAGS -O"
         need_asm=true
       fi
       ;;
-    *-*-linux*)
-      AC_MSG_RESULT(-fPIC)
-      PICFLAG=-fPIC
-      # Any Linux compiler had better be gcc compatible.
-      ;;
     *)
-      AC_MSG_RESULT("<none>")
       ;;
   esac
-fi
 CFLAGS="$CFLAGS -DNDEBUG"
 
-AC_SUBST(PICFLAG)
 AC_SUBST(DEFS)
 
 AM_CONDITIONAL(NEED_ASM, test x$need_asm = xtrue)
--- origsrc/libatomic_ops-1.2/src/Makefile.am	2005-03-10 22:34:22.000000000 -0600
+++ src/libatomic_ops-1.2/src/Makefile.am	2007-07-31 10:15:11.174902200 -0500
@@ -1,16 +1,17 @@
 SUBDIRS=atomic_ops
 
-AM_CFLAGS=@PICFLAG@
-
 include_HEADERS=atomic_ops.h atomic_ops_stack.h atomic_ops_malloc.h
-lib_LIBRARIES = libatomic_ops.a libatomic_ops_gpl.a
+lib_LTLIBRARIES = libatomic_ops.la libatomic_ops_gpl.la
 if NEED_ASM
-libatomic_ops_a_SOURCES = atomic_ops.c atomic_ops_sysdeps.S
+libatomic_ops_la_SOURCES = atomic_ops.c atomic_ops_sysdeps.S
 else
-libatomic_ops_a_SOURCES = atomic_ops.c
+libatomic_ops_la_SOURCES = atomic_ops.c
 endif
+libatomic_ops_la_LDFLAGS = -no-undefined
 
-libatomic_ops_gpl_a_SOURCES = atomic_ops_stack.c atomic_ops_malloc.c
+libatomic_ops_gpl_la_SOURCES = atomic_ops_stack.c atomic_ops_malloc.c
+libatomic_ops_gpl_la_LDFLAGS = -no-undefined
+libatomic_ops_gpl_la_LIBADD  = libatomic_ops.la
 
 EXTRA_DIST=Makefile.msft
 
